### PR TITLE
Tidy up javaharness WORKSPACE rules

### PR DESCRIPTION
**Pull request by csilvestrini**
Migrated from maven_jar to maven_install, and removed some unused j2cl
deps.

In a follow-up PR I'll combine the src_kt/WORKSAPCE file with the main
WORKSPACE file to make things even simpler.

### CODE REVIEW

File: WORKSPACE [hunk 1]
```diff
@@ -80,92 +42,42 @@ http_archive(
...
-    name = "autovalue",
-    artifact = "com.google.auto.value:auto-value:" + AUTO_VALUE_VERSION,
-    sha1 = "fe8387764ed19460eda4f106849c664f51c07121",
-)
-
-maven_jar(
-    name = "autovalue_annotations",
-    artifact = "com.google.auto.value:auto-value-annotations:" + AUTO_VALUE_VERSION,
-    sha1 = "5be124948ebdc7807df68207f35a0f23ce427f29",
+maven_install(
```
Comment (cromwellian) [inline WORKSPACE]: yay, I was just about to do this (have a version of it in my particle accelerator branch), now I don't have to split it out into separate PRs :)

File: javaharness/java/arcs/android/BUILD [hunk 2]
```diff
@@ -10,22 +10,24 @@ load("@build_bazel_rules_android//android:rules.bzl", "android_library")
...
         "//shells/pipes-shell:pipes_shell_web",
     ],
-    exports_manifest = 1,
     assets_dir = "pipes_shell_web_dist",
+    exports_manifest = 1,
+    idl_import_root = ".",
+    idl_srcs = glob(["*.aidl"]),
+    javacopts = ["-Xep:AndroidJdkLibsChecker:OFF"],
+    manifest = "AndroidManifest.xml",
+    plugins = [
```
Comment (cromwellian) [inline javaharness/java/arcs/android/BUILD]: Why is this now needed? Didn't it used to get picked up automatically fromthe dagger library?

File: WORKSPACE [hunk 3]
```diff
@@ -1,73 +1,35 @@
...
+    sha256 = "cd06d15dd8bb59926e4d65f9003bfc20f9da4b2519985c27e190cddc8b7a7806",
+    strip_prefix = "rules_android-0.1.1",
+    urls = ["https://github.com/bazelbuild/rules_android/archive/v0.1.1.zip"],
 )
 
-
+# Kotlin
 
 rules_kotlin_version = "legacy-modded-1_0_0-01"
+
```
[the remark this answers is not in the document]
Comment (csilvestrini) [reply WORKSPACE]: Done

File: javaharness/java/arcs/android/BUILD (hunk 2 above)
Comment (csilvestrini) [reply javaharness/java/arcs/android/BUILD]: Apparently not. I'm not sure how this used to work, but it was producing build errors now.

File: WORKSPACE (hunk 3 above)
[the remark this answers is not in the document]
Comment (cgruber) [reply WORKSPACE]: Can you confirm that it's also broken in this way for you? I'm having a real difficulty replicating this, and having more cases would help.

File: javaharness/java/arcs/android/BUILD (hunk 2 above)
Comment (cgruber) [reply javaharness/java/arcs/android/BUILD]: Does the java library for dagger have an `exported_plugins=` statement? If so, then this `plugins=` shouldn't be needed.

Comment (cgruber) [reply javaharness/java/arcs/android/BUILD]: Oh, I see. You started using rules_jvm_external, which doesn't automatically setup the dagger plugin system. You should make a dagger library somewhere which imports the core dagger jar, declares a `java_plugin(name = "dagger_plugin", ...)` for the dagger compiler, and have that core dagger library have an `exports_plugins=[":dagger_plugin"]`

Then, you depend on that local definition of dagger, instead of the `@maven//:com_google_dagger_dagger`.

**Comment by cgruber** [comment]
Sorry, I realize I'm commenting after the merge, but you might find it useful for a subsequent PR.
